unit/solvers/smt2_incremental/smt_commands.cpp

```diff
@@ -94,3 +95,15 @@ TEST_CASE("smt_set_option_commandt getter", "[core][smt2_incremental]")
   CHECK(smt_set_option_commandt{models_on}.option() == models_on);
   CHECK(smt_set_option_commandt{models_off}.option() == models_off);
 }
+
+TEST_CASE("SMT2 function application factory tests", "[core][smt2_incremental]")
```

Comment:
No `CHECK`s in this test? Whats its purpose?

Reply:
The checks were missing because I got distracted adding other functionality to this PR, before I finished writing this test. Thanks for spotting this oversight. I have now added some checks.